components/brave_wallet/browser/network_manager.cc

```diff
@@ -1070,15 +1059,15 @@ std::vector<mojom::NetworkInfoPtr> NetworkManager::GetAllChains() {
   return result;
 }
 
-std::optional<bool> NetworkManager::IsEip1559Chain(
-
-    std::string_view chain_id) {
+bool NetworkManager::IsEip1559Chain(std::string_view chain_id) {
   auto chain_id_lwr = base::ToLowerASCII(chain_id);
```

Comment:
I would prefer if we didn't have this string conversion here because:

 * This causes an allocation
 * It makes the code slightly harder to comprehend, as it is not clear why this conversion is necessary in the first place. Is it the case that we have unvalidated input providing these values? Not clear.
 * We may end up multiple conversions like those in different places, as it is impossible to determine if the conversion was already done.

Reply:
I've added a lot of these in all places where the lowercase conversion is happening, so we can monitor if this is absolutely necessary:

```
  if (!IsLowerCase(*chain_id)) {
    // TODO(https://github.com/brave/brave-browser/issues/46940): Adding these
    // dumps in all places where this conversion is being done in this file to
    // better understand why this conversion is required in the first place, and
    // if we can completely eliminate them.
    base::debug::DumpWithoutCrashing()
        << "Chain ID is not lowercase: " << *chain_id;
  }
```

Reply:
Actually I'm improving this a bit with `MakeChainIdLowerCase` to avoid hammering the application with crash reports.